### PR TITLE
Build options with machine type

### PR DESCRIPTION
**Pull request by guilhermearaujo**

### CODE REVIEW

File: flamingo/models/build.py [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Build(EmbeddedDocument):
     deploy_tag: str = None
     post_build_commands: List[str] = field(default_factory=list)
     build_args: KeyValue = field(default_factory=dict)
+    build_machine_type: str = None
```
Comment (joaodaher) [inline flamingo/models/build.py:21]: `Optional[str]`

**Review by joaodaher — APPROVED** [review]

File: flamingo/models/app.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ class App(Document):
     _environment: Environment = None
 
     def __post_init__(self):
-        environment = self.environment  # check if environment name actually exists, and caches it
+        _ = self.environment  # check if environment name actually exists, and caches it
```
Comment (guilhermearaujo) [inline flamingo/models/app.py:51]: Is this really needed?

Comment (joaodaher) [reply flamingo/models/app.py:51]: It caches the environment.

File: flamingo/services/alias_engine.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def replace(self, virtual_value):
         for alias_to in aliases_to:
             try:
                 replace_with = self.replacements[alias_to]
-                new_value = new_value.replace("${%s}" % alias_to, replace_with)
+                new_value = new_value.replace(f"${{alias_to}}", replace_with)
```
Comment (joaodaher) [inline flamingo/services/alias_engine.py]: Did you test this?
String formatting on Cloud Build triggers are **very** finicky.

Comment (guilhermearaujo) [reply flamingo/services/alias_engine.py]: Oops, indeed it was missing a pair of braces.

Now it should be ok:
```
>>> alias_to = "some_var"
>>> f"${{{alias_to}}}"
'${some_var}'
```

File: flamingo/models/build.py (hunk 1 above)
Comment (guilhermearaujo) [reply flamingo/models/build.py:21]: To use `Optional`, there's some heavy work that needs to be done on `gcp-pilot` first.
It crashes when using `issubclass` and some other things

**Pull request merged**